src/main/scala/firrtl/transforms/FlattenRegUpdate.scala

```diff
@@ -133,7 +134,7 @@ class FlattenRegUpdate extends Transform with DependencyAPIMigration {
   override def optionalPrerequisiteOf = Seq.empty
 
   override def invalidates(a: Transform): Boolean = a match {
-    case _: DeadCodeElimination => true
+    case _: DeadCodeElimination | ResolveKinds => true
```

Comment:
This can probably be fixed by adding the correct kind (`RegKind`) to the `WRef` created on line 100.